### ISSUE TITLE
Crash when exporting

### ISSUE THREAD

**Issue by screwtop** [hydrogen-music/hydrogen#1902]
Hydrogen version *          : 1.2.2- [Sep 10 2023] from OS repo
Operating system + version  : Arch Linux, kernel 6.6.2-arch1-1
Audio driver + version      : jackdmp 1.9.22

- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -

Hydrogen is crashing when I try to export to WAV (seemingly every time). However, it plays OK in realtime (so I can still record audio into other apps via JACK).

Here is the `hydrogen.log`:

```
Hydrogen 1.2.2- [Sep 10 2023]  [http://www.hydrogen-music.org]
Copyright 2002-2008 Alessandro Cominu
Copyright 2008-2023 The hydrogen development team
Hydrogen comes with ABSOLUTELY NO WARRANTY
This is free software, and you are welcome to redistribute it under certain conditions. See the file COPYING for details.
(E) ::static int H2Core::AudioEngine::audioEngine_process(uint32_t, void*) Failed to lock audioEngine in allowed 2.65567 ms, missed buffer
(E) ::static int H2Core::AudioEngine::audioEngine_process(uint32_t, void*) Failed to lock audioEngine in allowed 2.65567 ms, missed buffer
(E) ::static int H2Core::AudioEngine::audioEngine_process(uint32_t, void*) Failed to lock audioEngine in allowed 2.65567 ms, missed buffer
(E) ::static int H2Core::AudioEngine::audioEngine_process(uint32_t, void*) Failed to lock audioEngine in allowed 2.65567 ms, missed buffer
(E) ::static int H2Core::AudioEngine::audioEngine_process(uint32_t, void*) Failed to lock audioEngine in allowed 2.65567 ms, missed buffer
(E) ::static int H2Core::AudioEngine::audioEngine_process(uint32_t, void*) Failed to lock audioEngine in allowed 2.65567 ms, missed buffer
(E) ::static int H2Core::AudioEngine::audioEngine_process(uint32_t, void*) Failed to lock audioEngine in allowed 2.65567 ms, missed buffer
terminate called without an active exception
(E) ::void handleFatalSignal(int) Fatal signal 6
(E) ::void handleFatalSignal(int) /usr/bin/hydrogen(+0x225a92) [0x561dec26da92]
(E) ::void handleFatalSignal(int) /usr/lib/libc.so.6(+0x3e710) [0x7fd169e5c710]
(E) ::void handleFatalSignal(int) /usr/lib/libc.so.6(+0x8e83c) [0x7fd169eac83c]
(E) ::void handleFatalSignal(int) /usr/lib/libc.so.6(raise+0x18) [0x7fd169e5c668]
(E) ::void handleFatalSignal(int) /usr/lib/libc.so.6(abort+0xd7) [0x7fd169e444b8]
(E) ::void handleFatalSignal(int) /usr/lib/libstdc++.so.6(+0x9ca6f) [0x7fd16a09ca6f]
(E) ::void handleFatalSignal(int) /usr/lib/libstdc++.so.6(+0xb011c) [0x7fd16a0b011c]
(E) ::void handleFatalSignal(int) /usr/lib/libstdc++.so.6(+0xb0189) [0x7fd16a0b0189]
(E) ::void handleFatalSignal(int) /usr/lib/libstdc++.so.6(__gxx_personality_v0+0x23d) [0x7fd16a0af9cd]
(E) ::void handleFatalSignal(int) /usr/lib/libgcc_s.so.1(+0x1a6f4) [0x7fd16bea56f4]
(E) ::void handleFatalSignal(int) /usr/lib/libgcc_s.so.1(_Unwind_ForcedUnwind+0x130) [0x7fd16bea5e20]
(E) ::void handleFatalSignal(int) /usr/lib/libc.so.6(+0x95187) [0x7fd169eb3187]
(E) ::void handleFatalSignal(int) /usr/lib/libc.so.6(+0x8ae9b) [0x7fd169ea8e9b]
(E) ::void handleFatalSignal(int) /usr/lib/libc.so.6(+0x3e710) [0x7fd169e5c710]
(E) ::void handleFatalSignal(int) /usr/lib/libc.so.6(+0x89446) [0x7fd169ea7446]
(E) ::void handleFatalSignal(int) /usr/lib/libc.so.6(+0x903eb) [0x7fd169eae3eb]
(E) ::void handleFatalSignal(int) /usr/lib/libhydrogen-core-1.2.2.so(_ZN6H2Core11AudioEngine10tryLockForENSt6chrono8durationIlSt5ratioILl1ELl1000000EEEEPKcjS7_+0x8d) [0x7fd16bc7dc9d]
(E) ::void handleFatalSignal(int) /usr/lib/libhydrogen-core-1.2.2.so(_ZN6H2Core11AudioEngine19audioEngine_processEjPv+0xe5) [0x7fd16bc89d05]
(E) ::void handleFatalSignal(int) /usr/lib/libjack.so.0(+0x1444d) [0x7fd16a9c144d]
(E) ::void handleFatalSignal(int) /usr/lib/libjack.so.0(+0x13be8) [0x7fd16a9c0be8]
(E) ::void handleFatalSignal(int) /usr/lib/libjack.so.0(+0x31bb1) [0x7fd16a9debb1]
(E) ::void handleFatalSignal(int) /usr/lib/libc.so.6(+0x8c9eb) [0x7fd169eaa9eb]
(E) ::void handleFatalSignal(int) /usr/lib/libc.so.6(+0x1107cc) [0x7fd169f2e7cc]
```

**Comment by theGreatWhiteShark** [comment]
Hey @screwtop ,

Could upload the [log](https://github.com/hydrogen-music/hydrogen/wiki/Accessing-log-messages-of-Hydrogen) messages produced while the crash is happening and the .h2song file you used?

**Comment by theGreatWhiteShark** [comment]
Could you check whether the crash still occurs in [this version](https://ci.appveyor.com/api/buildjobs/u6io05ii3l0121ff/artifacts/Hydrogen-1.2.2-96-gc0cb947f4-x86_64.AppImage) of Hydrogen?

**Comment by theGreatWhiteShark** [comment]
Fixed in #1909
